src/Plugin/DecoderPlugin.php

```diff
@@ -123,18 +133,28 @@ private function decodeOnEncodingHeader($headerName, ResponseInterface $response
      */
     private function decorateStream($encoding, StreamInterface $stream)
     {
+        $decorated = null;
+        
         if ('chunked' === strtolower($encoding)) {
-            return new Encoding\DechunkStream($stream);
+            $decorated = new Encoding\DechunkStream($stream);
         }
 
         if ('deflate' === strtolower($encoding)) {
-            return new Encoding\DecompressStream($stream);
+            $decorated = new Encoding\DecompressStream($stream);
         }
 
         if ('gzip' === strtolower($encoding)) {
-            return new Encoding\GzipDecodeStream($stream);
+            $decorated = new Encoding\GzipDecodeStream($stream);
+        }
+        
+        if ($decorated === null) {
+            return null;   
+        }
+        
+        if ($this->buffered) {
+            $decorated = new BufferedStream($decorated);
```

Comment:
In fact this is a bad place as we only need this buffer once at the end, if we have a stream that is both gzip and chunked encoded we will have 2 buffered streams :/